rocAL/include/meta_data/meta_data.h

```diff
@@ -135,18 +142,50 @@ class MetaData {
     MetaDataInfo _info;
 };
 
+class AsciiValue : public MetaData {
+   public:
+    AsciiValue() = default;
+    AsciiValue(AsciiValues ascii_values) {
+        _ascii_values = std::move(ascii_values);
+    }
+    void set_ascii_values(AsciiValues ascii_values) override { _ascii_values = std::move(ascii_values); }
+    AsciiValues& get_ascii_values() override { return _ascii_values; }
+    // Not Implemnented
+    void set_labels(Labels label_ids) override { THROW("Not Implemented") }
+    std::vector<int>& get_labels() override {  THROW("Not Implemented") }
+    BoundingBoxCords& get_bb_cords() override { THROW("Not Implemented") }
+    void set_bb_cords(BoundingBoxCords bb_cords) override{THROW("Not Implemented")} 
+    std::vector<int>& get_polygon_count() override{THROW("Not Implemented")} 
+    std::vector<std::vector<int>>& get_vertices_count() override{THROW("Not Implemented")} 
+    MaskCords& get_mask_cords() override { THROW("Not Implemented") }
+    void set_mask_cords(MaskCords mask_cords) override { THROW("Not Implemented") }
+    void set_polygon_counts(std::vector<int> polygon_count) override { THROW("Not Implemented") }
+    void set_vertices_counts(std::vector<std::vector<int>> vertices_count) override{THROW("Not Implemented")} 
+    JointsData& get_joints_data() override { THROW("Not Implemented") }
+    void set_joints_data(JointsData* joints_data) override { THROW("Not Implemented") }
+
+   protected:
+    AsciiValues _ascii_values = {};  // For storing ASCII data only
+};
+
 class Label : public MetaData {
    public:
     Label(int label) { _label_ids = {label}; }
     Label() { _label_ids = {-1}; }
     std::vector<int>& get_labels() override { return _label_ids; }
     void set_labels(Labels label_ids) override { _label_ids = std::move(label_ids); }
     BoundingBoxCords& get_bb_cords() override { THROW("Not Implemented") }
-    void set_bb_cords(BoundingBoxCords bb_cords) override{THROW("Not Implemented")} std::vector<int>& get_polygon_count() override{THROW("Not Implemented")} std::vector<std::vector<int>>& get_vertices_count() override{THROW("Not Implemented")} MaskCords& get_mask_cords() override { THROW("Not Implemented") }
+    void set_bb_cords(BoundingBoxCords bb_cords) override{THROW("Not Implemented")} 
```

Comment:
why are we introducing so many functions that are not implemented?

Reply:
The functions are implemented as pure virtual functions in MetaData class so every function has to be overriden and specified as not implemented. In this particular instance, we have made a formatting change by moving multiple function declaration in a single line to a single line for each function